metrics/files/grafana/custom_metrics.js

```diff
@@ -0,0 +1,228 @@
+/* global _ */
+/*
+ * Grafana Scripted Dashboard to:
+ *
+ * Global accessable variables
+ * window, document, $, jQuery, ARGS, moment
+ *
+ * Return a dashboard object, or a function
+ *
+ * For async scripts, return a function, this function must take a single callback function,
+ * call this function with the dasboard object
+ *
+ * Author: Mike Pountney, Infracode Ltd
+ *
+ * Heavily inspired by work of: Anatoliy Dobrosynets, Recorded Future, Inc.
+ *
+ */
+
+// accessable variables in this scope
+var window, document, ARGS, $, jQuery, moment, kbn;
+
+var arg_from = '2h';
+
+var arg_title = "Custom Metrics";
+var arg_refresh = "1m";
+var arg_no_help = false;
+var arg_targets = "";
+
+if(!_.isUndefined(ARGS.no_help)) {
+  arg_no_help = ARGS.no_help;
+}
+
+if(!_.isUndefined(ARGS.title)) {
+  arg_title = ARGS.title;
+}
+
+if(!_.isUndefined(ARGS.refresh)) {
+  arg_refresh = ARGS.refresh;
+}
+
+if(!_.isUndefined(ARGS.targets)) {
+  arg_targets = ARGS.targets;
+}
+
+function get_all_metrics() {
+  var metrics_url = window.location.protocol + '//' + window.location.hostname.replace(/^grafana/,"graphite") + (window.location.port ? ":" + window.location.port : "") + '/metrics/index.json';
+  var res = [];
+  var req = new XMLHttpRequest();
+  req.open('GET', metrics_url, false);
+  req.send(null);
+  var obj = JSON.parse(req.responseText);
+  return obj;
+};
+
+//---------------------------------------------------------------------------------------
+
+function panel_help_text() {
+  var help_md = "### How to use this dashboard\n" +
+                "\n" +
+                "This dashboard expects a comma separated list of graphite target paths.\n" +
+                "\n" +
+                "These will typically need to be URI encoded.\n" +
+                "\n" +
+                "The graphs will then be displayed in order, as individual graphs.\n" +
+                "\n" +
+                "Arguments:\n" +
+                "\n" +
+                "* `no_help` -- omit this panel\n" +
+                "* `targets={target1},{target2},...`\n" +
+                "* `refresh={interval}` override default refresh interval of `1min`\n" +
```

Comment:
Oh or does the panel_all_metrics give you the links you'd need? Is this dashboard linked from the other dashboards in any way?